src/Raven.Server/Documents/Handlers/Processors/Subscriptions/SubscriptionsHandlerProcessorForCreateSubscription.cs

```diff
@@ -0,0 +1,79 @@
+    DocumentsOperationContext>
+{
+    public SubscriptionsHandlerProcessorForCreateSubscription([NotNull] SubscriptionsHandler requestHandler) : base(requestHandler)
+    {
+    }
+    protected override Task<ChangeVectorValidationResult> TryValidateChangeVectorAsync(SubscriptionsHandler requestHandler,
+        TransactionOperationContext context, SubscriptionCreationOptions options, SubscriptionConnection.ParsedSubscription sub)
+    {
+        return Task.FromResult(TryValidateChangeVector(requestHandler, context, options, sub));
+    }
+
+    protected override Task<(long Index, long SubscriptionId)> CreateSubscriptionAsync(SubscriptionsHandler requestHandler, string databaseName, long? id, bool? disabled, SubscriptionCreationOptions options,
+        ChangeVectorValidationResult changeVectorValidationResult)
+    {
+        return PutSubscriptionAndInitChangeVectorAsync(requestHandler, databaseName, id, disabled, options, changeVectorValidationResult);
+    }
+
+    protected override ValueTask GetResponsibleNodesAndWaitForExecutionAsync(SubscriptionsHandler requestHandler, string name, long index)
+    {
+        return GetResponsibleNodesAndWaitForExecution(requestHandler, name, index);
+    }
+
+    protected override void ThrowOnRevisionsIfNeeded(bool subRevisions)
+    {
+        // no-op
+    }
+
+    internal static ChangeVectorValidationResult TryValidateChangeVector(SubscriptionsHandler requestHandler, TransactionOperationContext context, SubscriptionCreationOptions options, SubscriptionConnection.ParsedSubscription sub)
+    {
+        if (Enum.TryParse(options.ChangeVector, out Constants.Documents.SubscriptionChangeVectorSpecialStates changeVectorSpecialValue))
+        {
+            switch (changeVectorSpecialValue)
+            {
+                case Constants.Documents.SubscriptionChangeVectorSpecialStates.BeginningOfTime:
+                    return new ChangeVectorValidationResult() { ChangeVectorsCollection = null, InitialChangeVector = null };
+                case Constants.Documents.SubscriptionChangeVectorSpecialStates.LastDocument:
+                    return new ChangeVectorValidationResult()
+                    {
+                        ChangeVectorsCollection = null,
+                        InitialChangeVector = requestHandler.Database.DocumentsStorage.GetLastDocumentChangeVector(context.Transaction.InnerTransaction, context, sub.Collection)
+                    };
+            }
+        }
+
+        return new ChangeVectorValidationResult() { ChangeVectorsCollection = null, InitialChangeVector = options.ChangeVector };
+    }
+
+    public static Task<(long Index, long SubscriptionId)> PutSubscriptionAndInitChangeVectorAsync(SubscriptionsHandler requestHandler, string databaseName, long? id, bool? disabled, SubscriptionCreationOptions options, ChangeVectorValidationResult changeVectorValidationResult)
```

Comment:
also here I think we can reduce the number of methods by removing the `static`

Reply:
you cannot since it used for both non sharding processors: `SubscriptionsHandlerProcessorForUpdateSubscription` & `SubscriptionsHandlerProcessorForCreateSubscription`